pybv/io.py

```diff
@@ -228,19 +259,36 @@ def _write_vmrk_file(vmrk_fname, eeg_fname, events, meas_date):
                   f'{i_dur},0', file=fout)
 
 
-def _scale_data_to_unit(data, unit):
-    """Scale `data` in Volts to `data` in `unit`."""
-    scale = SUPPORTED_UNITS.get(unit, None)
-    if scale is None:
-        msg = (f'Encountered unsupported unit: {unit}\n'
-               f'Use one of the following: {set(SUPPORTED_UNITS.keys())}')
-        raise ValueError(msg)
-
-    return data * scale
+def _scale_data_to_unit(data, units):
+    """Scale `data` in Volts to `data` in `units`."""
+    # keep track of any unsupported units to warn user
+    unsupported_units = set()
+
+    # create a vector to multiply with to play nice with numpy
+    scales = np.zeros((len(units), 1))
+    for idx, unit in enumerate(units):
+        scale = SUPPORTED_UNITS.get(unit, None)
+        # unless the unit is 'µV', then
+        # technically it is not supported in BrainVision
+        if unit != 'µV':
+            unsupported_units.add(unit)
+        # if not voltage unit at all, then don't scale
+        if scale is None:
+            scale = 1  # don't scale the data at all
+        scales[idx] = scale
+
+    if len(unsupported_units) > 0:
+        msg = (f'Encountered unsupported units: '
+               f'{", ".join(unsupported_units)}\n'
+               f'If this is a mistake, '
+               f'use one of the following: '
+               f'{", ".join(sorted(SUPPORTED_UNITS.keys()))}')
+        warn(msg)
```

Comment:
Could we differentiate between truly unsupported units (which we don't scale) and internally supported (but officially unsupported) units (i.e. `V`, `mV`, `nV`) and issue a different warning message? We do scale those volts in contrast to truly unsupported units.

Reply:
Done with a different message.